clang/lib/DPCT/BarrierFenceSpaceAnalyzer.cpp

```diff
@@ -356,65 +372,71 @@ const ArraySubscriptExpr *getArraySubscriptExpr(const DeclRefExpr *Node) {
   return ASE;
 }
 
-const Expr *getIdxOfASEConstValueExpr(const ArraySubscriptExpr *ASE) {
+// Match pattern:
+// CompoundAssignOperator '+='
+// |-DeclRefExpr
+// `-ImplicitCastExpr <LValueToRValue>
+//   `-DeclRefExpr
+bool isIncPattern(const DeclRefExpr *DRE, std::string &RHSStr) {
+  const CompoundAssignOperator *CO =
+      DpctGlobalInfo::findParent<CompoundAssignOperator>(DRE);
+  if (!CO)
+    return false;
+  if (CO->getOpcode() != BinaryOperatorKind::BO_AddAssign)
+    return false;
+  const DeclRefExpr *RHS =
+      dyn_cast<DeclRefExpr>(CO->getRHS()->IgnoreImpCasts());
+  if (!RHS)
+    return false;
+  RHSStr = ExprAnalysis::ref(RHS);
+  return true;
+}
+
+std::tuple<const Expr *, bool, std::string>
+getIdxOfASEConstValueExpr(const ArraySubscriptExpr *ASE) {
```

Comment:
Function body has similar code in `trackInitExprOfDRE(const DeclRefExpr *DRE)`. Can we merge them?

Reply:
Keep as different functions currently.